url2698a2c731_cassandra_project_cassandra_stand_alone_tgz-pJ8-eu_cassandra_sim_MySimpleDynamicScenarioSimulationJ8/src/eu/cassandra/sim/entities/appliances/Appliance.java

```diff
@@ -0,0 +1,432 @@
+		// Optional or state related variables
+		private long onTick = -1;
+		private String who = null;
+		
+		/**
+		 * Instantiates a new builder, with required only parameters.
+		 *
+		 * @param aid the appliance id
+		 * @param aname the appliance name
+		 * @param adesc the appliance description
+		 * @param atype the appliance type
+		 * @param ainstallation the installation the appliance belongs to
+		 * @param apcm the  active power consumption model of the appliance
+		 * @param aqcm the  reactive power consumption model of the appliance
+		 * @param astandy the stand-by consumption  of the appliance
+		 * @param abase whether the appliance is a base load
+		 */
+		public Builder(String aid, String aname, String adesc, String atype,
+				Installation ainstallation, ConsumptionModel apcm,
+				ConsumptionModel aqcm, double astandy, boolean abase) {
+			id = aid;
+			name = aname;
+			description = adesc;
+			type = atype;		
+			installation = ainstallation;
+			pcm = apcm;
+			qcm = aqcm;
+			standByConsumption = astandy;
+			base = abase;
+		}
+		
+		/**
+		 * Builds the Appliance.
+		 *
+		 * @param orng the random number generator
+		 * @return the appliance
+		 */
+		public Appliance build(ORNG orng) {
+			return new Appliance(this, orng);
+		}
+	}
+
+	/**
+	 * Instantiates a new appliance.
+	 *
+	 * @param builder the builder
+	 * @param orng the random number generator
+	 */
+	private Appliance(Builder builder, ORNG orng) {
+		id = builder.id;
+		name = builder.name;
+		description = builder.description;
+		type = builder.type;
+		installation = builder.installation;
+		standByConsumption = builder.standByConsumption;
+		pcm = builder.pcm;
+		qcm = builder.qcm;
+		base = builder.base;
+		inUse = (base) ? true : false;
+		onTick = (base) ? -orng.nextInt(Constants.MIN_IN_DAY) : builder.onTick;
+		who = builder.who;
+	}
+
+	/**
+	 * Gets the installation the appliance belongs to.
+	 *
+	 * @return the installation the appliance belongs to
+	 */
+	public Installation getInstallation() {
+		return installation;
+	}
+
+	/**
+	 * Checks whether the appliance is in use.
+	 *
+	 * @return whether the appliance is in use
+	 */
+	public boolean isInUse() {
+		return inUse;
+	}
+
+	/**
+	 * Checks whether the appliance is a base load.
+	 *
+	 * @return whether the appliance is a base load
+	 */
+	public boolean isBase() {
+		return base;
+	}
+
+	/**
+	 * Gets the reactive power consumption model.
+	 *
+	 * @return the reactive power consumption model
+	 */
+	public ConsumptionModel getQConsumptionModel() {
+		return qcm;
+	}
+
+	/**
+	 * Gets the active power consumption model.
+	 *
+	 * @return the active power consumption model
+	 */
+	public ConsumptionModel getPConsumptionModel() {
+		return pcm;
+	}
+
+	/**
+	 * Gets the power.
+	 *
+	 * @param tick the tick
+	 * @param type the type
+	 * @return the power
+	 */
+	public double getPower(long tick, String type) {
+		try {
+
+			ConsumptionModel cm = null; 
+			if(type == "p") {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An equality operator (== or !=) is used to compare strings, which matches strings based on address.

**Fix**: Do content comparison using the equals() method to compare the values of the strings.